core/src/test/java/fr/sncf/osrd/envelope_physics/TrainPhysics.java

```diff
@@ -0,0 +1,123 @@
+package fr.sncf.osrd.envelope_physics;
+
+import static fr.sncf.osrd.envelope_sim.TrainPhysicsIntegrator.getWeightForce;
+import static fr.sncf.osrd.envelope_sim.TrainPhysicsIntegrator.newtonStep;
+import static org.junit.jupiter.api.Assertions.*;
+
+import fr.sncf.osrd.envelope_sim.Action;
+import fr.sncf.osrd.envelope_sim.EnvelopeSimContext;
+import fr.sncf.osrd.envelope_sim.FlatPath;
+import fr.sncf.osrd.envelope_sim.TrainPhysicsIntegrator;
+import fr.sncf.osrd.train.RollingStock;
+import fr.sncf.osrd.train.TestTrains;
+import org.junit.jupiter.api.Test;
+
+
+public class TrainPhysics {
+
+    RollingStock testRollingStock = TestTrains.REALISTIC_FAST_TRAIN;
+    double timeStep = 1.0;
+
+    @Test
+    public void testSlopeNoTraction() {
+        var testPath = new FlatPath(100000, -40);
+        var context = new EnvelopeSimContext(testRollingStock, testPath, timeStep);
+        double position = 0.0;
+        double speed = 0.0;
+
+        // how fast would a train go after 10 steps of 1 sec, coasting on a
+        // -40m / km slope?
+        for (int i = 0; i < 10; i++) {
+            var step = TrainPhysicsIntegrator.step(context, position, speed, Action.COAST, +1);
+            position += step.positionDelta;
+            speed = step.endSpeed;
+        }
+
+        // we expect about +4m/s (the train goes forward)
+        assertTrue(speed < +5 && speed > +1, String.valueOf(speed));
```

Comment:
I'm actually thinking maybe calculating on a longer time and have a more constraining assert ? Expecting 4m/s between 5 and 1 is a bit soft i think.
I know this is how the test was previously built but i'm saying we can improve it as we port in to the new integration system :)

Reply:
So maybe calculate on someting like 30s or 60s and assert the speed with a relatively more constraining error margin